src/precice/tests/DataContextTest.cpp

```diff
@@ -55,23 +55,76 @@ BOOST_AUTO_TEST_CASE(testDataContextWriteMapping)
+  // Create the object for to mesh and the data
+  mesh::PtrMesh ptrToMesh2 = std::make_shared<mesh::Mesh>("SecondOtherMesh", dimensions, testing::nextMeshID());
+  mesh::PtrData ptrToData2 = ptrToMesh2->createData("MappedData", dimensions, 2_dataID);
+
+  ptrToMesh2->createVertex(Eigen::Vector3d(0.0, 1.1, 0.0));
+  ptrToMesh2->createVertex(Eigen::Vector3d(2.0, 1.1, 0.0));
+  ptrToMesh2->createVertex(Eigen::Vector3d(0.0, 2.1, 4.0));
+
+  MeshContext toMeshContext2(dimensions);
+  toMeshContext2.mesh = ptrToMesh2;
+
+  MappingContext mappingContext2;
+  mappingContext2.fromMeshID = ptrFromMesh->getID();
+  mappingContext2.toMeshID   = ptrToMesh2->getID();
+
+  // the mapping configuration
+  dataContext.addMappingConfiguration(mappingContext2, toMeshContext2);
+
+  // First, we repeat the checks from above in order to check that nothing changed
+  BOOST_TEST(fixture.hasMapping(dataContext));
+  BOOST_TEST(fixture.getProvidedDataID(dataContext) != ptrToData->getID());
+  BOOST_TEST(fixture.getProvidedDataID(dataContext) == ptrFromData->getID());
+  BOOST_TEST(dataContext.getMeshID() != ptrToMesh->getID());
+  BOOST_TEST(dataContext.getMeshID() == ptrFromMesh->getID());
+  BOOST_TEST(fixture.hasWriteMapping(dataContext));
+  BOOST_TEST(!fixture.hasReadMapping(dataContext));
+  // Test dedicated content of the first mapping configuration
+  BOOST_TEST(fixture.getFromDataID(dataContext, 0) == ptrFromData->getID());
+  BOOST_TEST(fixture.getToDataID(dataContext, 0) == ptrToData->getID());
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].fromMeshID == mappingContext.fromMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].toMeshID == mappingContext.toMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].hasMappedData == mappingContext.hasMappedData);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].mapping == mappingContext.mapping);
+  BOOST_TEST(fixture.mappingContext(dataContext)[0].timing == mappingContext.timing);
+
+  // Now, test the newly added mapping
+  BOOST_TEST(fixture.getProvidedDataID(dataContext) != ptrToData2->getID());
+  BOOST_TEST(dataContext.getMeshID() != ptrToMesh2->getID());
+  BOOST_TEST(fixture.hasWriteMapping(dataContext));
+  BOOST_TEST(!fixture.hasReadMapping(dataContext));
+  // Test dedicated content of the first mapping configuration
+  BOOST_TEST(fixture.getFromDataID(dataContext, 1) == ptrFromData->getID());
+  BOOST_TEST(fixture.getToDataID(dataContext, 1) == ptrToData2->getID());
+  BOOST_TEST(fixture.mappingContext(dataContext)[1].fromMeshID == mappingContext2.fromMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[1].toMeshID == mappingContext2.toMeshID);
+  BOOST_TEST(fixture.mappingContext(dataContext)[1].hasMappedData == mappingContext2.hasMappedData);
+  BOOST_TEST(fixture.mappingContext(dataContext)[1].mapping == mappingContext2.mapping);
+  BOOST_TEST(fixture.mappingContext(dataContext)[1].timing == mappingContext2.timing);
 }
 
 BOOST_AUTO_TEST_CASE(testDataContextReadMapping)
```

Comment:
Do we also need a test for multiple read mappings?

Reply:
No, this would be ambiguous from the user perspective and it is also ensured via an assertion.